aiidalab_widgets_base/viewers.py

```diff
@@ -221,34 +225,36 @@ def highlight_atoms(self,
         self._viewer._remove_representations_by_name(repr_name='selected_atoms')  # pylint:disable=protected-access
         self._viewer.add_ball_and_stick(  # pylint:disable=no-member
             name="selected_atoms",
-            selection=vis_list,
+            selection=vis_list or [],
             color=color,
             aspectRatio=size,
             opacity=opacity)
 
-    def _apply_selection(self, change=None):  # pylint:disable=unused-argument
-        """Apply selection specified in the text area."""
-        short_selection = change['new']
-        expanded_selection, syntax_ok = string_range_to_set(short_selection)
-        self.set_trait('selection', expanded_selection)
-
-        with self.hold_trait_notifications():
-            if syntax_ok:
-                self.wrong_syntax.layout.visibility = 'hidden'
-                self.highlight_atoms(self.selection)
-            else:
-                self.wrong_syntax.layout.visibility = 'visible'
+    @default('selection')
+    def _default_selection(self):  # pylint: disable=no-self-use
+        return set()
 
-    def clear_selection(self, change=None):  # pylint:disable=unused-argument
+    @validate('selection')
+    def _validate_selection(self, change):
+        """If selection is provided as string, convert it into set."""
```

Comment:
Since we are converting here anyways, there is no need to allow the `selection` value to be Unicode, correct?

Reply:
sure, it is fine.